packages/fiori-annotation-api/src/cds/writer.ts

```diff
@@ -1416,3 +1399,26 @@ function processNode(content: ContainerContentBlock[], item: Comment | AstNode):
         content.push(element);
     }
 }
+function skipCommaInsertion(
+    changes: CDSDocumentChange[],
+    content: ContainerContentBlock[],
+    document: CDSDocument,
+    insertAfterIndex: number
+) {
+    return !!changes.find((change) => {
+        if (!change.type.startsWith('delete')) {
+            return false;
+        }
+        const astNodes = getAstNodesFromPointer(document, change.pointer);
+        const toBeDeletedNodes = astNodes[astNodes.length - 1];
+        const node = content[insertAfterIndex];
+        if (node.type === 'element' && node?.element === toBeDeletedNodes) {
+            if (toBeDeletedNodes.type === 'annotation' && content.length === 1) {
```

Comment:
This condition is for following case
https://github.com/SAP/open-ux-tools/blob/47c1b25e04ff418a09ab9bbc5ea378d41edbf4b2/packages/fiori-annotation-api/test/unit/cds/writer.test.ts#L611

Reply:
Here the target is coverted to annotationGroup after deletion

Reply:
That is compound annotation. Just not to mix it up with the annotation group AST node which groups multiple annotations from the same vocabulary.